### PR TITLE
Improve txs execution

### PR DESCRIPTION
**Pull request by SebastianMarian**
* Resized the max gas limit in one mini block for better balancing cross-shard and in-shard txs processing
* Increased the max number of pending mini blocks per shard allowance, after which a trigger for shard is stuck is activated

### CODE REVIEW

File: integrationTests/vm/esdt/common.go [hunk 1]
```diff
@@ -3,6 +3,7 @@ package esdt
 import (
 	"encoding/hex"
 	"fmt"
+	"github.com/ElrondNetwork/elrond-go/config"
```
Comment (iulianpascalau) [inline integrationTests/vm/esdt/common.go:6]: go imports

File: process/economics/economicsData.go [hunk 2]
```diff
@@ -709,7 +709,7 @@ func (ed *economicsData) setGasLimitConfig(currentEpoch uint32) {
 		"minGasLimit", ed.minGasLimit,
 	)
 
-	ed.statusHandler.SetUInt64Value(common.MetricMaxGasPerTransaction, core.MinUint64(ed.maxGasLimitPerMiniBlock, ed.maxGasLimitPerMetaMiniBlock))
+	ed.statusHandler.SetUInt64Value(common.MetricMaxGasPerTransaction, core.MaxUint64(process.MaxGasLimitPerTxForSafeCrossShard, core.MinUint64(ed.maxGasLimitPerMiniBlock, ed.maxGasLimitPerMetaMiniBlock)))
```
Comment (iulianpascalau) [inline process/economics/economicsData.go:712]: this is a little hard to understand for someone not involved in the discussions.

File: integrationTests/multiShard/txScenarios/common.go [hunk 3]
```diff
@@ -2,6 +2,7 @@ package txScenarios
 
 import (
 	"fmt"
+	"github.com/ElrondNetwork/elrond-go/config"
```
[the remark this answers is not in the document]
Comment (SebastianMarian) [reply integrationTests/multiShard/txScenarios/common.go:5]: done

File: integrationTests/vm/esdt/common.go (hunk 1 above)
Comment (SebastianMarian) [reply integrationTests/vm/esdt/common.go:6]: done

File: process/economics/economicsData.go (hunk 2 above)
Comment (SebastianMarian) [reply process/economics/economicsData.go:712]: I will add a comment in all the places where this calculation is used

File: process/block/preprocess/basePreProcess.go [hunk 4]
```diff
@@ -329,6 +329,8 @@ func (bpp *basePreProcess) computeGasConsumed(
 	if bpp.shardCoordinator.SelfId() == senderShardId {
 		gasConsumedByTxInSelfShard = gasConsumedByTxInSenderShard
 
+		// the max gas limit to be compared with, should be the maximum value between max gas limit per mini block and max gas limit per tx. If max gas limit per tx
```
Comment (iulianpascalau) [inline process/block/preprocess/basePreProcess.go:332]: does this comment apply here?

File: process/coordinator/process.go [hunk 5]
```diff
@@ -1300,6 +1300,8 @@ func (tc *transactionCoordinator) checkGasConsumedByMiniBlockInReceiverShard(
 		}
 	}
 
+	// the max gas limit to be compared with, should be the maximum value between max gas limit per mini block and max gas limit per tx. If max gas limit per tx
```
Comment (iulianpascalau) [inline process/coordinator/process.go:1303]: does this comment apply here?

File: process/block/preprocess/basePreProcess.go (hunk 4 above)
Comment (SebastianMarian) [reply process/block/preprocess/basePreProcess.go]: Partially yes, I have adjusted the comment a little bit

File: process/coordinator/process.go (hunk 5 above)
Comment (SebastianMarian) [reply process/coordinator/process.go]: Partially yes, I have adjusted the comment a little bit

**Review by iulianpascalau — COMMENTED** [review]
Looks good :+1:

**Review by sasurobert — DISMISSED** [review]

**Review by AdoAdoAdo — APPROVED** [review]

**Review by iulianpascalau — COMMENTED** [review]
looks good :+1:

**Review by AdoAdoAdo — DISMISSED** [review]

**Review by sasurobert — APPROVED** [review]

**Review by AdoAdoAdo — APPROVED** [review]

**Review by AdoAdoAdo — COMMENTED** [review]
should we close this PR?
As far as I know the changes have been merged in development in a different PR
@SebastianMarian

**Pull request closed without merging**